### DISCUSSION TITLE
`n_lags` Lagged regressor to `n_forecasts`  forecast steps ratio

### DISCUSSION THREAD

**Discussion by ghost** [ourownstory/neural_prophet#320] [Q&A - forecasting best practices] [answered]
What is the optimal ratio when using n_forecasts and n_lags?

Should n_lags always be greater than n_forecasts?

What does n_lag really mean? Lagging by n rows?

**Comment by ourownstory** [comment, marked as answer]
@veioxve3r1d23seb2g6b 

> What does n_lag really mean? Lagging by n rows?

`n_lags` is the number of most recent observations to include in the model. e.g. `n_lag=3` uses all of the last 3 hours as input to the model. 

> Should n_lags always be greater than n_forecasts?

There is no rule set in stone. Generally, it should be at least same.


> What is the optimal ratio when using n_forecasts and n_lags?

The conventional ratio is around 1:1 to 1:2, sometimes higher.
e.g. hourly data: when predicting next 6h, use last 24h
e.g. hourly data: when predicting next 72h, use last 72h
